pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -204,26 +204,36 @@ func convertServices(cfg config.Config) []*model.Service {
 	}
 
 	return buildServices(hostAddresses, cfg.Name, cfg.Namespace, svcPorts, serviceEntry.Location, resolution,
-		exportTo, labelSelectors, serviceEntry.SubjectAltNames, creationTime, cfg.Labels)
+		exportTo, labelSelectors, serviceEntry.SubjectAltNames, creationTime, cfg.Labels, clusterID)
 }
 
 func buildServices(hostAddresses []*HostAddress, name, namespace string, ports model.PortList, location networking.ServiceEntry_Location,
 	resolution model.Resolution, exportTo map[visibility.Instance]bool, selectors map[string]string, saccounts []string,
-	ctime time.Time, labels map[string]string,
+	ctime time.Time, labels map[string]string, clusterID cluster.ID,
 ) []*model.Service {
 	out := make([]*model.Service, 0, len(hostAddresses))
 	lbls := labels
 	if features.CanonicalServiceForMeshExternalServiceEntry && location == networking.ServiceEntry_MESH_EXTERNAL {
 		lbls = ensureCanonicalServiceLabels(name, labels)
 	}
 	for _, ha := range hostAddresses {
-		out = append(out, &model.Service{
+		clusterVIPs := []string{ha.address}
+		if features.EnableDualStack && ha.address == constants.UnspecifiedIP {
+			// Make sure to add the IPV6 unspecified address as well
+			clusterVIPs = append(clusterVIPs, constants.UnspecifiedIPv6)
```

Comment:
Hi @keithmattix, it's hard to detect IP families for external of services which are imported in Mesh, such service entry which has only hostname, no other information. In this situation, I am prefer to follow the proxy's ip families which need to create listener for this kind of service.
If we add both of the unspecified ips, besides the unit tests, many places will be impacted since `ha.address` only be set to `constants.UnspecifiedIP` at the beginning of the converting.

Reply:
Ah you're referring to ipv6 only proxies? That's a good call out; we probably shouldn't assume the proxy can support both. In that case, I _think_ we could move this logic into GetAddressesForProxy when generating the listeners so that we have the target proxy info at that point. WDYT?

Reply:
OK yeah this new approach is much less invasive. I'll validate locally and push it up